Linkedlist.java

```diff
@@ -0,0 +1,117 @@
+package src;
+
+import java.util.Scanner;
+
+public class Linkedlist
+{
+
+      void printlinkedlist(Node head)
+       {
+          Node current;
+          current=head;
+           while(current!=null)
+          {
+            System.out.println(current.data);
+            current=current.next;
+          }
+       }
+      
+      //To calculate size of the linked list
+      int length(Node head)
+      {
+         int count=0;
+         Node temp;
+         temp=head;
+         while(temp!=null)
+         {
+          count++;
+          temp=temp.next;
+         }
+         return count;
+      }
+
+      //Reverse the linkedlist
+      void reverse(Node head)
+      {
+         Node p=head,q=head;
+         int len;
+         int i,j,temp;
+         int k;
+         i=0;
+         len=length(head);
+         j=len-1;
+         while(i<j)
+         {
+           k=0;
+           while(k<j)
+           {
+             q=q.next;
+             k++;
+           }
+
+           temp=p.data;
+           p.data=q.data;
+           q.data=temp;
+
+           i++;
+           j--;
+        
+           p=p.next;
+           q=head;
+         }
+      }
+
+      //Search and print second last element of the Linkedlist
+      void second_last(Node head)
+      {   
+         Node current;
+         current=head;
+         while(current.next.next!=null)
+         {
+           current=current.next;
+         }
+         System.out.println(current.data);
+      }   
+
+      
+      public static void main(String[] args)
```

Comment:
Taking user input is not the functionality of LinkList move it out of this class